SmartTutor/ContentModel/LearningObjects/ChallengeModel/FulfillmentStrategy/NameChecker/NamingRule.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace SmartTutor.ContentModel.LearningObjects.ChallengeModel.FulfillmentStrategy.NameChecker
+{
+    public class NamingRule
+    {
+        [Key] public int Id { get; set; }
+        public List<string> BannedWords { get; set; }
+        public List<string> RequiredWords { get; set; }
+        public ChallengeHint Hint { get; set; }
+
+        internal ChallengeHint Evaluate(string name)
+        {
+            return NameMeetRequirements(name) ? null : Hint;
+        }
+
+        private bool NameMeetRequirements(string name)
+        {
+            return CheckNameLength(name) && CheckNameContent(name);
+        }
+
+        private bool CheckNameLength(string name)
+        {
+            return name.Length >= 2 && name.Length <= 35;
```

Comment:
This should be configurable as well through the field.

Reply:
I allowed tutor to set min and max length for words, but left default values if he chooses not to. This looks to me like a better option than constant fields. If I'm wrong let me know.